Source/Core/Core/State.cpp

```diff
@@ -398,6 +384,35 @@ static void CompressAndDumpState(CompressAndDumpState_args save_args)
     f.WriteBytes(buffer_data, buffer_size);
   }
 
+  {
+    std::lock_guard lk(s_save_thread_mutex);
+
+    // Backup existing state (overwriting an existing backup, if any).
+    if (File::Exists(filename))
+    {
+      if (File::Exists(File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav"))
+        File::Delete((File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav"));
+      if (File::Exists(File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav.dtm"))
+        File::Delete((File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav.dtm"));
+
+      if (!File::Rename(filename, File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav"))
+        Core::DisplayMessage("Failed to move previous state to state undo backup", 1000);
+      else if (File::Exists(filename + ".dtm"))
```

Comment:
Minor but I see there are a lot of `filename + ".dtm"`.  Maybe save in a variable to avoid concatenation cost each time:

`const auto dtm_filename = fmt::format("{}.dtm", filename);`

Reply:
Yeah I just copy-pasted that entire block and didn't think about it much, but you're right, this can be improved.